go.mod

```diff
@@ -2,6 +2,8 @@ module github.com/celestiaorg/celestia-app/v4
 
 go 1.23.6
 
+toolchain go1.24.3
```

Comment:
[question] can we reconcile these two versions and bump the `go ` version to 1.24.3 if we need the toolchain bumped to 1.24.3?

Also if we do this, can we bump the README?

Reply:
removed, the tooling is a bit annoying as it just auto added these